test/coretasks/test_coretasks_sasl.py

```diff
@@ -253,6 +262,51 @@ def test_sasl_plain_no_password(
     ), 'No password is a configuration error and the bot must quit.'
 
 
+def test_sasl_external_no_password(
+    botfactory: BotFactory,
+    configfactory: ConfigFactory,
+) -> None:
+    tmpconfig = configfactory("conf.ini", TMP_CONFIG_SASL_EXTERNAL)
+    mockbot = botfactory.preloaded(tmpconfig, preloads=["coretasks"])
+    mockbot.backend.connected = True
+
+    # connect and capability negotiation
+    mockbot.on_connect()
+    mockbot.on_message(":irc.example.com CAP * LS :sasl=PLAIN,EXTERNAL")
+    n = len(mockbot.backend.message_sent)
+    mockbot.on_message(":irc.example.com CAP * ACK :sasl")
+    mockbot.on_message("AUTHENTICATE +")
+    mockbot.on_message(":irc.example.com 903 SopelTest :SASL authentication successful")
+    assert mockbot.backend.message_sent[n:] == rawlist(
+        "AUTHENTICATE EXTERNAL",
+        "AUTHENTICATE +",
+        "CAP END",
+    ), "SASL EXTERNAL authentication should continue without a password set."
```

Comment:
This need to be a bit more thorough in its testing of step orders: it is important to check that we first have the 2 `AUTHENTICATE` messages, then the `CAP END`, because Sopel **must not** send the `CAP END` before the `903` (SASL sucess) event is received from the server.

```suggestion
    mockbot.on_message(":irc.example.com CAP * ACK :sasl")
    mockbot.on_message("AUTHENTICATE +")
    assert mockbot.backend.message_sent[n:] == rawlist(
        "AUTHENTICATE EXTERNAL",
        "AUTHENTICATE +",
    ), "SASL EXTERNAL authentication should continue without a password set."
    n = n + 2
    # assume it all went well
    mockbot.on_message(":irc.example.com 903 SopelTest :SASL authentication successful")
    assert mockbot.backend.message_sent[n:] == rawlist(
        "CAP END",
    ), "SASL success must resume capability negotiation"
```